crates/egui/src/text_selection/visuals.rs

```diff
@@ -58,7 +58,7 @@ pub fn paint_text_selection(
 
         // Start by appending the selection rectangle to end of the mesh, as two triangles (= 6 indices):
         let num_indices_before = mesh.indices.len();
-        mesh.add_colored_rect(rect, color);
+        mesh.add_colored_rect(rect, color.gamma_multiply(0.5));
```

Comment:
I think 0.5 might be a bit much. Maybe we should also add this as a style option, e.g. `highlight_color`?